src/mappings/entities/mapSettlement.ts

```diff
@@ -48,34 +48,40 @@ const instructionResultMap = {
 export const createLeg = async (
   blockId: string,
   instructionId: string,
+  address: string,
   legIndex: number,
   { ticker, amount, from, to }: LegDetails
 ): Promise<void> =>
   Leg.create({
     id: `${instructionId}/${legIndex}`,
-    ticker,
+    assetId: ticker,
     amount,
     fromId: `${from.identityId}/${from.number}`,
     toId: `${to.identityId}/${to.number}`,
     instructionId,
+    addresses: [address],
     createdBlockId: blockId,
     updatedBlockId: blockId,
   }).save();
 
 const updateLegs = async (
   blockId: string,
+  address: string,
   instructionId: string,
-  settlementId: string
-): Promise<void> => {
+  settlementId?: string
+): Promise<Promise<void>[]> => {
   const legs = await Leg.getByInstructionId(instructionId);
 
-  await Promise.all(
-    legs.map(leg => {
+  return legs.map(leg => {
```

Comment:
If this did  `return Promise.all(legs.map(leg => ....)` I think the function return type would be `Promise<void[]>` which looks neater than having a nested promise